lgc/patch/NggPrimShader.cpp

```diff
@@ -5070,11 +5097,16 @@ Function *NggPrimShader::createSmallPrimFilterCuller(Module *module) {
     //
     // Small primitive filter culling algorithm is described as follow:
     //
-    //   if ((roundEven(min(scaled(x0/w0), scaled(x1/w1), scaled(x2/w2))) ==
-    //        roundEven(max(scaled(x0/w0), scaled(x1/w1), scaled(x2/w2)))) ||
-    //       (roundEven(min(scaled(y0/w0), scaled(y1/w1), scaled(y2/w2))) ==
-    //        roundEven(max(scaled(y0/w0), scaled(y1/w1), scaled(y2/w2)))))
-    //       cullFlag = true
+    //   if (roundEven(min(screen(x0/w0), screen(x1/w1), screen(x2/w2)) ==
+    //       roundEven(max(screen(x0/w0), screen(x1/w1), screen(x2/w2))) ||
+    //       roundEven(min(screen(y0/w0), screen(y1/w1), screen(y2/w2)) ==
+    //       roundEven(max(screen(y0/w0), screen(y1/w1), screen(y2/w2))))
+    //     cullFlag = true
+    //
+    //   if !((w0 < 0 && w1 < 0 && w2 < 0) || (w0 > 0 && w1 > 0 && w2 > 0))
+    //     cullFlag = false
+    //
+    //   cullFlag = conservativeRaster ? false : cullFlag
```

Comment:
Perhaps we should instead skip the entire block if `conservativeRaster` is true? That is, `or i1` of conservativeRaster with the incoming cullFlag for the initial conditional branch.

Reply:
Ah. Good catch. We do need early exit.